withastro/starlight
# Fix type inference for extended i18n schemas

#### Description

- Closes #1787 
- Fixes type inference for `labels` when a user provides additional translation strings by extending Starlight’s i18n schema
- Previously these were statically typed based on our default schema shape. Now these should infer from the content collection type if configured.

#### Testing

To test this, I followed these manual steps:

1. Extended the i18n schema in `docs/src/content/config.ts` with an extra field:

   ```ts
   schema: i18nSchema({ extend: z.object({ test: z.string() }) }),
   ```
   
2. I added a test file at `docs/src/content/i18n/lang.yml` to make Astro detect the collection, and added some content:

   ```yml
   test: test
   ```
   
3. I created a test component using our overrides prop type to see if it would pick up the extended field in labels:

    ```astro
    ---
    import type { Props } from '@astrojs/starlight/props';

    Astro.props.labels.test
    //                 ^ string
    ---
    ```
